### PR TITLE
Add README describing use of semantic convention YAML models

### LINKED ISSUE #979: Add documentation to the semantic convention YAML models directory
I found the YAML files in the semantic_conventions directory at the root of this repository to be something of a mystery. When I first discovered them, I thought they were duplicates of the semantic conventions found in markdown in the specs. It took quite a bit of digging to find this  [pr](https://github.com/open-telemetry/opentelemetry-specification/pull/571) and [issue](https://github.com/open-telemetry/opentelemetry-specification/issues/547) and subsequently [this build tool](https://github.com/open-telemetry/build-tools/tree/master/semantic-conventions), and now I've used the docker build tool to generate semantic convention markdown.

I propose that we add a README to the main directory of the semantic conventions with links out to the build tools, so the purpose of these files will be more obvious to newcomers.

### PR DESCRIPTION
**Pull request by justinfoote**
Fixes #979

## Changes

This PR doesn't change any existing files, it only adds a new README to the semantic_conventions directory at the root of this repo.
I've verified that the addition of this readme does not affect the functioning of the semantic convention build tool.

### CODE REVIEW

**Review by thisthat — APPROVED** [review]
Thank you for clarifying this :)

File: semantic_conventions/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# YAML Model for Semantic Conventions
+
+The YAML descriptions of semantic convention contained in this directory are intended to
+be used by the various language OpenTelemetry language implementations to aid in automatic
```
Comment (ThomsonTan) [inline semantic_conventions/README.md:4]: Duplicated "language" word?

File: semantic_conventions/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+generation of semantics-related code.
+
+## Generation
+
+Using these YAML models, and the [semantic convention generator](https://github.com/open-telemetry/build-tools/tree/master/semantic-conventions)
```
Comment (Oberon00) [inline semantic_conventions/README.md:9]: Please mention `make table-generation` https://github.com/open-telemetry/opentelemetry-specification/blob/3e66e6fcc3bb8070db7ccbdd30005ae9f35ac29c/Makefile#L40 (which should be renamed to `make generate-tables` @thisthat)

File: semantic_conventions/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+consistently-formatted Markdown tables for all of our semantic conventions,
+and to generate code for use in OpenTelemetry language projects.
+
+See also:
+* [Markdown Tables](https://github.com/open-telemetry/build-tools/tree/master/semantic-conventions#markdown-tables)
+* [Code Generator](https://github.com/open-telemetry/build-tools/tree/master/semantic-conventions#code-generator)
+
+## Description of the model
+
+The fields and their expected values are presented in [allowed_syntax.md](./allowed_syntax.md).
```
Comment (Oberon00) [inline semantic_conventions/README.md:20]:
```suggestion
The fields and their expected values are presented in [syntax.md](./syntax.md).
```

File: semantic_conventions/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+For a basic attribute description, refer to the [General Attributes example](./trace/general.yaml).
```
Comment (Oberon00) [inline semantic_conventions/README.md:22]:
```suggestion
For a basic attribute description, refer to the [General Attributes](./trace/general.yaml).
```

It's not an example, it's actual conventions.

File: semantic_conventions/README.md (hunk 2 above)
Comment (justinfoote) [reply semantic_conventions/README.md]: I've updated it to reference the current makefile `table-generation` target. 
Should I also change the target name? (This feels like a bigger change that might affect people's workflows, and will definitely require CI changes, right?)

File: semantic_conventions/README.md (hunk 4 above)
Comment (justinfoote) [reply semantic_conventions/README.md]: Good point. Fixed in 8ebc19c.

File: semantic_conventions/README.md (hunk 3 above)
Comment (justinfoote) [reply semantic_conventions/README.md]: Fixed in 8ebc19c.

File: semantic_conventions/README.md (hunk 1 above)
Comment (justinfoote) [reply semantic_conventions/README.md]: Fixed in 8ebc19c.

**Review by tigrannajaryan — APPROVED** [review]

**Review by arminru — APPROVED** [review]
Thanks!

**Review by carlosalberto — APPROVED** [review]

**Pull request merged**